src/Microsoft.Health.Dicom.Core/Features/Store/StoreOrchestrator.cs

```diff
@@ -115,4 +144,92 @@ private async Task TryCleanupInstanceIndexAsync(VersionedInstanceIdentifier vers
             // Fire and forget.
         }
     }
+
+    internal static Dictionary<int, FrameRange> GetFramesOffset(DicomDataset dataset)
```

Comment:
Should these logics go in DicomFilExtensions?

Reply:
Good thought, this is very specific to Store, so would like to keep it here instead of exposing more.